syft/format/cyclonedxxml/encoder.go

```diff
@@ -0,0 +1,78 @@
+package cyclonedxxml
+
+import (
+	"github.com/CycloneDX/cyclonedx-go"
+
+	"github.com/anchore/syft/syft/format/internal/cyclonedxutil"
+	"github.com/anchore/syft/syft/sbom"
+)
+
+var _ sbom.FormatEncoder = (*encoder)(nil)
+
+const ID = cyclonedxutil.XMLFormatID
+
+func SupportedVersions() []string {
+	return cyclonedxutil.SupportedVersions(ID)
+}
+
+type EncoderConfig struct {
+	Version string
+}
+
+type encoder struct {
+	cfg EncoderConfig
+	cyclonedxutil.Encoder
+}
+
+func NewFormatEncoder(cfg EncoderConfig) (sbom.FormatEncoder, error) {
+	enc, err := cyclonedxutil.NewEncoder(cfg.Version, cyclonedx.BOMFileFormatXML)
+	if err != nil {
+		return nil, err
+	}
+	return encoder{
+		cfg:     cfg,
+		Encoder: enc,
+	}, nil
+}
+
+func DefaultFormatEncoder() sbom.FormatEncoder {
+	enc, err := NewFormatEncoder(DefaultEncoderConfig())
+	if err != nil {
+		panic(err)
+	}
+	return enc
+}
+
+func DefaultFormatEncoders() []sbom.FormatEncoder {
```

Comment:
All of the `DefaultEncoder` functions have been removed.

Is there a strong reason to remove the `DefaultEncoderConfig` functions? I could make them unexported, but they are used across packages so this would get rather annoying (I'd need to make a top-level internal package for this).

Reply:
I think leaving the defaults for _configs_ is definitely fine!